VisualizationBase/src/nodes/InfoNode.cpp

```diff
@@ -52,18 +52,32 @@ InfoNode::InfoNode(Model::Node *target, QJsonArray enabledInfos)
 	Q_ASSERT(target);
 	for (auto item : enabledInfos)
 		setEnabled(item.toString(), true);
-	allInfoNodes.append(this);
+	initialize();
 }
 
 InfoNode::~InfoNode()
 {
-	allInfoNodes.removeAll(this);
+	allInfoNodes.remove(key_);
 	target_ = nullptr;
 }
 
+void InfoNode::initialize()
```

Comment:
Not for very much - essentially for allowing registering new functions in the InfoNode to affect existing InfoNodes, which already existed before (just as a list, not a hash). But this isn't really needed anywhere right now.